File: halcyon-web/src/main/resources/static/js/controller/hr/hrEmployeeController.js
//控制层
app.controller("hrEmployeeController", function ($scope, $controller, hrEmployeeService, hrPositionService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.hrEmployee = {};
        $scope.getHrPositionOptions();

    }
    $scope.initForUpdate = function (id) {
        hrEmployeeService.queryById(id).success(function (data) {
            $scope.hrEmployee = data;
        });
        $scope.getHrPositionOptions();
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        hrEmployeeService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.save = function () {
        hrEmployeeService.save($scope.hrEmployee).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    }

    function deleteRows() {
        hrEmployeeService.delete($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }

    //获取下拉框数据
    //防止ajax异步导致为空，声明一个空的
    $scope.hrPositionOptions = {data: []};
    $scope.getHrPositionOptions = function () {
        hrPositionService.getParentPositionOptions().success(function (data) {
            $scope.hrPositionOptions = {data: data};
        });
    }
});

File: halcyon-base/src/main/java/com/marshal/halcyon/base/hr/controller/HrEmployeeController.java
package com.marshal.halcyon.base.hr.controller;

import com.marshal.halcyon.core.entity.ResponseData;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.base.hr.entity.HrEmployee;
import com.marshal.halcyon.base.hr.service.HrEmployeeService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * @auth: Marshal
 * @date: 2019/1/9
 * @desc:
 */
@RestController
@RequestMapping("/hr/employee")
public class HrEmployeeController extends BaseController {

    @Autowired
    HrEmployeeService hrEmployeeService;

    /**
     * 员工信息查询
     *
     * @param condition
     * @param pageNum
     * @param pageSize
     * @return
     */
    @RequestMapping("/query")
    public ResponseData query(@RequestBody HrEmployee condition, int pageNum, int pageSize) {
        List<HrEmployee> list = hrEmployeeService.select(condition, pageNum, pageSize);
        return new ResponseData(list);
    }

    /**
     * 部门信息提交
     *
     * @param hrEmployee
     * @return
     */
    @RequestMapping("/submit")
    public ResponseData submit(@RequestBody HrEmployee hrEmployee) {
        if (!getValidator().hasError(hrEmployee)) {
            return new ResponseData(false, getValidator().getErrors(hrEmployee));
        }
        hrEmployeeService.save(hrEmployee);
        return new ResponseData(true, "保存成功");
    }

    /**
     * 部门信息删除
     *
     * @param selectedIds
     * @return
     */
    @RequestMapping("/remove")
    public ResponseData remove(@RequestParam("selectedIds") Long[] selectedIds) {
        hrEmployeeService.batchDelete(selectedIds);
        return new ResponseData(true, "删除成功");
    }

    @RequestMapping("/selectByEmployeeId")
    public HrEmployee selectByEmployeeId(@RequestParam Long employeeId) {
        return hrEmployeeService.selectByPrimaryKey(employeeId);
    }

    @RequestMapping("/getEmpOptions")
    public List<Map> getEmpOptions() {
        return hrEmployeeService.getEmpOptions();
    }
}
